app/frontend/src/features/donations/DonationsBox.tsx

```diff
@@ -0,0 +1,481 @@
+import {
+  Divider,
+  FormControl,
+  FormControlLabel,
+  FormHelperText,
+  makeStyles,
+  Radio,
+  RadioGroup,
+  Typography,
+} from "@material-ui/core";
+import { loadStripe } from "@stripe/stripe-js";
+import classNames from "classnames";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import {
+  DONATIONSBOX_ALERT_SUCCESS,
+  DONATIONSBOX_ALERT_WARNING,
+  DONATIONSBOX_CURRENCY,
+  DONATIONSBOX_MONTHLY,
+  DONATIONSBOX_NEXT,
+  DONATIONSBOX_ONETIME,
+  DONATIONSBOX_RECURRING,
+  DONATIONSBOX_RECURRING_ARIA,
+  DONATIONSBOX_REQUIRED,
+  DONATIONSBOX_TEXT,
+  DONATIONSBOX_TITLE,
+  DONATIONSBOX_VALUES,
+} from "features/donations/constants";
+import { Error as GrpcError } from "grpc-web";
+import { useRef, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import { useLocation } from "react-router-dom";
+import { service } from "service";
+
+const useStyles = makeStyles((theme) => ({
+  donationsBox: {
+    padding: theme.spacing(2),
+    border: `2px solid ${theme.palette.grey[200]}`,
+    borderRadius: theme.shape.borderRadius * 2,
+  },
+
+  donationsBoxRow: {
+    gridTemplateColumns: "repeat( auto-fit, minmax(160px, 1fr) )",
+    gridAutoRows: "2.75rem",
+    display: "grid",
+    gridGap: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+
+  donationsBoxSubRow: {
+    gridTemplateColumns: "repeat( auto-fit, minmax(72px, 1fr) )",
+    display: "grid",
+    gridGap: theme.spacing(2),
+  },
+
+  buttonSecondary: {
+    "&&": {
+      boxShadow: "initial",
+      color: theme.palette.grey[600],
+      cursor: "pointer",
+      fontWeight: 700,
+      fontSize: theme.typography.button.fontSize,
+      transition: `color ${theme.transitions.duration.short}ms ${theme.transitions.easing.easeInOut}`,
+      justifyContent: "center",
+      alignItems: "center",
+      border: `2px solid ${theme.palette.grey[200]}`,
+      borderRadius: "0.5rem",
+      backgroundColor: theme.palette.grey[200],
+      margin: "initial",
+      height: "100%",
+      width: "100%",
+      display: "flex",
+    },
+
+    "&:hover": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      backgroundColor: theme.palette.background.paper,
+      color: theme.palette.primary.main,
+      transition: `color ${theme.transitions.duration.short} ${theme.transitions.easing.easeInOut}`,
+    },
+  },
+
+  buttonSecondaryRadio: {
+    "&&": {
+      boxShadow: "initial",
+      margin: 0,
+    },
+    "&:hover > .MuiFormControlLabel-label": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      backgroundColor: theme.palette.background.paper,
+      color: theme.palette.primary.main,
+      transition: `color ${theme.transitions.duration.short} ${theme.transitions.easing.easeInOut}`,
+    },
+    "& > .MuiRadio-root": {
+      position: "fixed",
+      top: "-99999px",
+      left: "-99999px",
```

Comment:
```suggestion
```
Don't think this is necessary once opacity is already at 0 (presumably so keyboard arrows between them still works), since `position: fixed` already takes the radio button out of the normal document flow and so won't impact the rest of the "button"

Reply:
...but it has the drawback that the "real" radio button would probably still be somewhere in the viewport, so any user could accidentally click this invisible button when trying to click something else, which is not desirable imo.

Reply:
I don't think it's that big of a deal, since the radio is still within the bounds of the button when I looked. This looks quite hacky and shifts the focus on visual users only, which isn't a very inclusive solution

Reply:
https://webaim.org/techniques/css/invisiblecontent/ suggests `position: absolute; left: -10000px;` so I will change it to that. It's probably 10ns more performant than 0 opacity.